LuaMenu/widgets/chobby/components/ai_list_window.lua

```diff
@@ -79,15 +80,35 @@ function AiListWindow:AddAiToList(ai, blackList, oldAiVersions, isRunning64Bit)
 		tooltip = tooltip,
 		OnClick = {
 			function()
-				self:AddAi(displayName, shortName, ai.version)
-				self:HideWindow()
+				local function defaultAction()
+					self:AddAi(displayName, shortName, ai.version)
+					self:HideWindow()
+				end
+
+				local skirmishDefault = WG.Chobby.Configuration.gameConfig.skirmishDefault
+				if not skirmishDefault or not skirmishDefault.AIOptionsEnabled then
+					defaultAction()
+					return
+				end
+
+				local path = "AI/Skirmish/" .. ai.shortName .. "/" .. ai.version .. "/AIOptions.lua"
```

Comment:
Add a lobby option for displaying AI options, because having an extra window pop up each time you want to add an AI sounds pretty annoying. Games should be able to set the default state of this option.

Reply:
@rlcevg would you be able to take a look at this?

Reply:
Already there, WG.Chobby.Configuration.gameConfig.skirmishDefault.AIOptionsEnabled
just 2 lines above this code.
Or are we talking about some GUI, then what's wrong with current option?

Reply:
I missed skirmishDefault because it is just meant to be a default. The file should only define the default state of skirmish rooms when players open it. It also isn't a setting, which would be a tickbox in the lobby menu.

Come to think of it, both a popup and a tickbox are fairly inelegant. An entry in the combobox menu for the AI entry in the playerlist would be much cleaner. Selecting the entry could bring up AIOptions for that AI.

Reply:
I can't visualize others text. Here's possible option
![aioptions](https://user-images.githubusercontent.com/514675/98440762-43bcd780-2103-11eb-8fb7-f0853a042dd2.png)
but i don't know gui framework, can it do "slide down" effect?
Or just 1st part with button (arrow) leading to new window with aioptions

Reply:
Also what are settings? Could you please link a related file so i can start look on how settings work and how to add one

Reply:
Search the repository for all instances of `showOldAiVersions`. This is settings, it handles the entire settings tab: https://github.com/ZeroK-RTS/Chobby/blob/master/LuaMenu/widgets/gui_settings_window.lua#L1050

I was suggesting than an option be added to the combobox (here https://github.com/ZeroK-RTS/Chobby/blob/master/LuaMenu/widgets/api_user_handler.lua#L152) for changing an AIs options. This would just bring up an options window, similar to what exists already in BAR. You could also make the options window optionally appear when the AI is selected by having the player click on an advanced button. A telescoping menu like the one in your picture would also be possible.

Reply:
Got it, will get back to it soon, right after "faction/side selector" PR applied.

Reply:
Done.